src/test/kotlin/pt/up/fe/ni/website/backend/controller/RoleControllerTest.kt

```diff
@@ -0,0 +1,720 @@
+        }
+    }
+
+    @NestedTest
+    inner class GrantPermissionRole {
+
+        @BeforeEach
+        fun addRole() {
+            roleRepository.save(testRole)
+        }
+
+        @Test
+        fun `should grant permission to role that exists`() {
+            mockMvc.perform(
+                post("/roles/${testRole.id}/grant")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(
+                        objectMapper.writeValueAsString(
+                            mapOf(
+                                "permissions" to Permissions(listOf(Permission.SUPERUSER))
+                            )
+                        )
+                    )
+            ).andExpectAll(
+                status().isOk
+            )
+                .andDocumentEmptyObjectResponse(
+                    documentationPermissions,
+                    "Adds a set of permissions to a role by ID",
+                    "This doesn't check if the permission is already added, the role ID must be valid"
+                )
+            assert(roleRepository.findByIdOrNull(testRole.id!!)!!.permissions.contains(Permission.SUPERUSER))
+        }
+
+        @Test
+        fun `should grant permission to role that does not exists`() {
+            mockMvc.perform(
+                post("/roles/1234/grant")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(
+                        objectMapper.writeValueAsString(
+                            mapOf(
+                                "permissions" to Permissions(listOf(Permission.SUPERUSER))
+                            )
+                        )
+                    )
+            ).andExpectAll(
+                status().isNotFound()
+            ).andDocumentErrorResponse(documentationPermissions, hasRequestPayload = false)
+        }
+    }
+
+    @NestedTest
+    inner class RevokePermissionRole {
+
+        @BeforeEach
+        fun addRole() {
+            roleRepository.save(testRole)
+        }
+
+        @Test
+        fun `should revoke permission to role that exists`() {
+            mockMvc.perform(
+                post("/roles/${testRole.id}/revoke")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(
+                        objectMapper.writeValueAsString(
+                            mapOf(
+                                "permissions" to Permissions(listOf(Permission.SUPERUSER))
+                            )
+                        )
+                    )
+            )
+                .andExpectAll(
+                    status().isOk
+                ).andDocumentEmptyObjectResponse(
+                    documentationPermissions,
+                    "Revokes permission by role ID",
+                    "Revokes permissions, it doesn't check if the permissions to be revoked are already off."
```

Comment:
```suggestion
                    "Revokes permissions, it doesn't check if the role contains them."
```
seems simpler, wdyt?